### PR TITLE
(feat) Integrate month-day-preview, integrate month with backend

### PR DESCRIPTION
**Pull request by joshuaNewman10**

### CODE REVIEW

File: client/js/monthData.js [hunk 1]
```diff
@@ -1,151 +1,48 @@
 /*Temporary Mock Month Data */
```
Comment (kyleaowen) [inline client/js/monthData.js]: are you using this data somewhere?

File: client/js/stores/day-store.js [hunk 2]
```diff
@@ -53,13 +54,12 @@ var DayStore = assign({}, EventEmitter.prototype, {
...
 Dispatcher.register(function(payload){
   switch (payload.actionType){
-    case AppConstants.ServerActionTypes.DAY_DATA_RECIEVED:
+    case AppConstants.ServerActionTypes.DAY_DATA_RECEIVED:
       _dayData.gigs = _.indexBy(payload.gigs, function(gig){
         return gig.id;
       });
       _dayData.date = payload.date;
       DayStore.emitChange();
-      break;
```
Comment (kyleaowen) [inline client/js/stores/day-store.js:62]: why did you remove this break?

File: client/js/stores/month-store.js [hunk 3]
```diff
@@ -14,6 +14,50 @@ var _setMonthData = function (data) {
...
+  for (var i=1; i<=numWeeks; i++) {
+    monthData.data[i] = {};
+  }
+
+  //Fill the day properties of each week as with 'YYYYMMDD' as the key
+  var dataKey = '';
+  currentDate = moment(data.startDate, 'YYYY-MM-DD');
+  for(var j=1; j<=numDays; j++) {
+    dataKey = currentDate.format('YYYYMMDD');
+    weekNum = Math.ceil(j / 7);
```
Comment (kyleaowen) [inline client/js/stores/month-store.js:45]: the variable weekNum is missing a declaration.

File: client/css/eventfull.css [hunk 4]
```diff
@@ -242,6 +242,11 @@ input[type=submit]{
 }
 
 .month-canvas {
+  padding-right: 0px;
+  margin-right: 0px;
+}
+
+.day-wrapper {
```
Comment (rjatkinson2) [inline client/css/eventfull.css]: What's going on with this class?

File: client/js/actions/server-action-creator.js [hunk 5]
```diff
@@ -21,9 +21,9 @@ var ServerActionCreator = {
       weekData: weekData
     });
   },
-  fetchedMonthData: function (monthData) {
+  receiveMonthData: function (monthData) {
```
Comment (rjatkinson2) [inline client/js/actions/server-action-creator.js:24]: Glad we are normalizing our verbiage.

File: client/js/components/month/month.js [hunk 6]
```diff
@@ -1,41 +1,43 @@
...
+var moment = require('moment');
 
 var Month = React.createClass({
   getInitialState: function () {
     return {
-      'week1': [{date: '', numEvents: 0, status: ''}],
-      'week2': [{date: '', numEvents: 0, status: ''}],
-      'week3': [{date: '', numEvents: 0, status: ''}],
-      'week4': [{date: '', numEvents: 0, status: ''}]
+      startDate : '2015-05-10',
```
Comment (rjatkinson2) [inline client/js/components/month/month.js]: What's the meaning of this start date? Should be able to just use the current day.

File: client/js/components/month/month.js [hunk 7]
```diff
@@ -1,41 +1,43 @@
...
+      date: null
     };
   },
   componentWillMount: function () {
+    var today = moment().startOf('month');
```
Comment (rjatkinson2) [inline client/js/components/month/month.js]: I don't think the variable name today make sense if it is assigned to the first day of the month?

File: client/js/components/month/month-prev-button.js [hunk 8]
```diff
@@ -1,19 +1,19 @@
...
-        <button className="month-prev" onClick={this.handleClick}>Prev Month</button>       
-      </div>
+      <div className="arrow arrow-left" onClick={this.handleClick}></div>
     );
   }
 });
 
-module.exports = MonthPrevButton;
\ No newline at end of file
+module.exports = MonthPrevButton;
```
Comment (rjatkinson2) [inline client/js/components/month/month-prev-button.js]: These two files (MonthNext and MonthPrev) are exactly the same except for the className and the date method. You could break this down to a single component and pass the className and method in as props.

File: client/js/components/month/month-header.js [hunk 9]
```diff
@@ -8,12 +8,12 @@ var MonthHeader = React.createClass({
   render: function () {
     return (
       <div className="date-bar">
-        <div className="arrow arrow-left"></div>
-        <h2>May 2015</h2>
-        <div className="arrow arrow-right"></div>
+        <PrevMonth className="arrow arrow-left" month={this.props.month} year={this.props.year} />
```
Comment (rjatkinson2) [inline client/js/components/month/month-header.js]: classNames should only exist on the render method of the component. if you want them to be dynamic, you could pass them in as props.

File: client/js/components/month/month-day-preview.js [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+    var date = this.state.date ? this.state.date : moment();
+    date = date.format('MMMM Do');
+    var data = [];
+    if ( this.state.gigs !== null ) {
+      data = _.map(this.state.gigs, function (gig, key) {
+        return gig;
+      });
+    }
+    return (
+      <div className = "col-md-3" onClick={ this.toggleDetails }>
```
Comment (rjatkinson2) [inline client/js/components/month/month-day-preview.js]: I don't see toggleDetails in this component?

File: client/js/components/month/month-day-event.js [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+var Link = require('react-router-component').Link;
+var moment = require('moment');
+var AppViewActions = require('../../actions/view-action-creator');
+
+var MonthDayEvent = React.createClass({
+  handleClick: function () {
+    var date = moment(this.props.eventData.date, 'YYYY-MM-DD');
+    AppViewActions.getDayData(date);
+  },
+  render: function () {
```
Comment (rjatkinson2) [inline client/js/components/month/month-day-event.js:12]: This is a bit repetitive. Maybe something like:

``` javascript
var color = this.props.health === 10 ? 'green' : this.props.health < 4 ? 'red' : 'blue';
return <div onClick={ this.handleClick } className={ 'events-circle' + color } ></div>;
```

File: client/js/components/month/month-day-card.js [hunk 12]
```diff
@@ -1,31 +1,25 @@
...
 var React = require('react');
+var MonthDayEvent = require('./month-day-event');
 var Link = require('react-router-component').Link;
+var moment = require('moment');
 
 var MonthDayCard = React.createClass({
   render:function () {
-    var currentDate = this.props.dayData.date;
-    var dayPath = '/day/' + currentDate;
+    var weekNumber = Number(this.props.dayData.weekNumber);
```
Comment (rjatkinson2) [inline client/js/components/month/month-day-card.js]: I don't see this used again.

File: client/js/components/month/month-day-card.js [hunk 13]
```diff
@@ -1,31 +1,25 @@
...
+    var dayNumber = parseInt(moment(this.props.dayData.dayNumber, 'YYYYMMDD').format('DD'), 10);
```
Comment (rjatkinson2) [inline client/js/components/month/month-day-card.js]: Why do you need to parseInt here?

File: client/css/eventfull.css (hunk 4 above)
Comment (joshuaNewman10) [inline client/css/eventfull.css]: Good catch! Removed!

File: client/js/components/month/month-day-card.js (hunk 12 above)
Comment (joshuaNewman10) [inline client/js/components/month/month-day-card.js]: Good catch, it was from an old version where I needed to pass in the week and day and do a calculation based on that

File: client/js/components/month/month-day-card.js (hunk 13 above)
Comment (joshuaNewman10) [inline client/js/components/month/month-day-card.js]: It was from when moment was causing problems and not converting to the correct day as expected, it will be removed in my corrections

File: client/js/components/month/month-day-event.js (hunk 11 above)
Comment (joshuaNewman10) [inline client/js/components/month/month-day-event.js:12]: Good call, much simpler!

File: client/js/components/month/month-day-preview.js [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+  },
+  _onChange: function () {
+    this.setState(DayStore.getData());
+  },
+  render: function () {
+    var date = this.state.date ? this.state.date : moment();
+    date = date.format('MMMM Do');
+    var data = [];
+    if ( this.state.gigs !== null ) {
+      data = _.map(this.state.gigs, function (gig, key) {
```
Comment (kyleaowen) [inline client/js/components/month/month-day-preview.js]: what is the purpose of this map? it seems like you could pass this.state.gigs directly into the WeekDayBin and it would work similarly

File: client/js/components/month/month-prev-button.js (hunk 8 above)
Comment (joshuaNewman10) [inline client/js/components/month/month-prev-button.js]: Cool, I thought it made more sense to break them out into two seperate components since they are pretty short and we may want to differentiate them in some fundamental way. But im happy to make them into one reusable component since that is what we have done for week and day

Comment (kyleaowen) [inline client/js/components/month/month-prev-button.js]: If you want to make an arrow header abstraction, allow to take the parameters "direction" and "timeSpan". direction can be "left" or "right", timeSpan can be "day" "month" or "week"

Comment (kyleaowen) [inline client/js/components/month/month-prev-button.js]: that may be something we could refactor later. for now, it may be easiest to just do something similar to the day-header, where the logic is all in one place

File: client/js/monthData.js (hunk 1 above)
Comment (joshuaNewman10) [inline client/js/monthData.js]: So if our server was down I wanted to render an empty calendar. But just talked to Ryan and we agreed that it makes more sense to render nothing. That way we won't mislead the users about there not being any events for a given day

its been added to gitignore

**Comment by kyleaowen** [comment]
I noticed that ViewActionCreator is sometimes being called AppViewActions. I think we should stick with ViewActionCreator, which is present across the rest of our codebase, thoughts?

File: client/js/monthData.js (hunk 1 above)
Comment (kyleaowen) [inline client/js/monthData.js]: is it being used as defaults somewhere? I'd imagine that if our server was down, we'd want to display an error message and ask users to check back later or something

File: client/js/components/month/month-nav-arrow.js [hunk 15]
```diff
@@ -0,0 +1,20 @@
+/** @jsx React.DOM */
+var React = require('react');
+var AppViewActions = require('../../actions/view-action-creator');
```
Comment (kyleaowen) [inline client/js/components/month/month-nav-arrow.js]: ViewActionCreator?

**Comment by rjatkinson2** [comment]
Thanks for rocking the changes!
